packages/config/src/VitessceAutoConfig.test.js

```diff
@@ -511,22 +526,468 @@ describe('src/VitessceAutoConfig.js', () => {
   });
 
   it('raises an error when URL with unsupported file format is passed', async () => {
     const urls = ['http://localhost:4204/@fixtures/zarr/anndata-0.7/somefile.zarr'];
 
-    await generateConfigs(urls).catch(
-      e => expect(e.message).toContain('This file type is not supported.'),
+    await generateConfig(urls, NO_HINTS_CONFIG, '', false).catch(
+      e => expect(e.message).toContain('One or more of the URLs provided point to unsupported file types.'),
     );
   });
+
+  // ********** TESTS WITH HINTS **********
+
+  it('generates config with hints for OME-TIFF dataset types correctly', async () => {
+    const urls = ['somefile.ome.tif'];
+    const datasetType = ['OME-TIFF'];
+    const expectedConfig = {
+      version: '1.0.15',
+      name: 'An automatically generated config. Adjust values and add layout components if needed.',
+      description: 'Populate with text relevant to this visualisation.',
+      datasets: [
+        {
+          uid: 'A',
+          name: 'An automatically generated view config for dataset. Adjust values and add layout components if needed.',
+          files: [
+            {
+              fileType: 'raster.json',
+              options: {
+                images: [
+                  {
+                    metadata: {
+                      isBitmask: false,
+                    },
+                    name: 'somefile.ome.tif',
+                    type: 'ome-tiff',
+                    url: urls[0],
+                  },
+                ],
+                schemaVersion: '0.0.2',
+                usePhysicalSizeScaling: false,
+              },
+            },
+          ],
+        },
+      ],
+      coordinationSpace: {
+        dataset: {
+          A: 'A',
+        },
+      },
+      layout: [
+        {
+          component: 'spatial',
+          coordinationScopes: {
+            dataset: 'A',
+          },
+          x: 0,
+          y: 0,
+          w: 8,
+          h: 12,
+        },
+        {
+          component: 'layerController',
+          coordinationScopes: {
+            dataset: 'A',
+          },
+          x: 8,
+          y: 0,
+          w: 4,
+          h: 7,
+          props: {
+            disable3d: [],
+            disableChannelsIfRgbDetected: true,
+          },
+        },
+        {
+          component: 'description',
+          coordinationScopes: {
+            dataset: 'A',
+          },
+          h: 5,
+          w: 4,
+          x: 8,
+          y: 9,
+        },
+      ],
+      initStrategy: 'auto',
+    };
+
+    const hintsConfig = getDatasetHintsConfig(urls);
+    const expectedHintsConfig = HINTS_CONFIG.find(c => c.hintType.length === 1
+      && c.hintType.includes(...datasetType));
+    expect(hintsConfig).toEqual(expectedHintsConfig);
+
+    const config = {
+      hintsType: datasetType,
+      ...expectedHintsConfig.hints[1],
+    };
+
+    const viewConfig = await generateConfig(urls, config);
```

Comment:
Is there a reason this logic can't be moved inside the `generateConfig` function? It would be preferable to make the function signature for `generateConfig` to be as straightforward as possible to make it amenable to future applications such as command-line usage where the user simply needs to specify the URLs and hint as a string

```
const viewConfig = await generateConfig(urls, 'Transcriptomics / scRNA-seq (with heatmap)');
```

an equivalent command-line usage could then be

```sh
./generate-config.sh --urls http://localhost/something.adata.zarr --hint 'Transcriptomics / scRNA-seq (with heatmap)'
```

There might then be cases in which `generateConfig` needs to throw errors if the urls and/or hint does not match anything in `HINTS_CONFIG` but that should be fine

Reply:
Really good point, Mark! I started realising that too after working on the CellBrowser-to-Vitessce config conversion issue. I have refactored the code now to support only supplying a string with the hint name when calling `generateConfig` function.